lib/redactor.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+# This is a first-pass, low-tech approach to redacting PII from our logs.
+# We will eventually move this to a gem so that our other Rails apps can leverage the code.
+# We may also want to use gems such as 'redactor' or 'ruby_regex'.
+module Redactor
+  PATTERNS = {
+    ssn: /\b\d{3}[-\sxX._]?\d{2}[-\sxX._]?\d{4}\b/,
+    email: /[a-zA-Z0-9][-a-zA-Z0-9._]+@[-a-zA-Z0-9]+\.+[a-zA-Z]{2,8}/,
+    cc: /\b(?:\d[ -.]*?){13,17}\b/,
+    phone: /\(?([2-9]\d{2})\)?[- .]?([2-9]\d{2}[- .]?\d{4})/
+  }.freeze
+
+  def self.redact(str)
+    str = str.dup
+    PATTERNS.each { |key, value| str.gsub!(value, "[redacted_#{key}]") }
```

Comment:
I tried hard to find a more elegant way to do this, but it gets the job done. I confirmed that the performance impact is negligible when compared to raw logging:
```ruby
> Benchmark.realtime { 10000.times { Rails.logger.info("[Search Impression] #{Redactor.redact(hash.to_json)}") } }
0.8479499999666587
> Benchmark.realtime { 10000.times { Rails.logger.info("[Search Impression] #{hash.to_json}") } }
0.7438500000862405
```